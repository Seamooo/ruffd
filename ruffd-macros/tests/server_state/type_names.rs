use ruffd_macros::server_state;
use ruffd_types::tokio::sync::RwLock;
use std::sync::Arc;

#[server_state]
pub struct Foo {
    pub foo: u32,
    pub bar: bool,
}

fn main() {
    let new_state = Foo {
        foo: Arc::new(RwLock::new(3)),
        bar: Arc::new(RwLock::new(false)),
    };
    let locks = FooLocks::default();
    let handles_fut = foo_handles_from_locks(locks);
}
