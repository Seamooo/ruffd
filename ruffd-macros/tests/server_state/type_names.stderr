error[E0308]: mismatched types
  --> tests/server_state/type_names.rs:17:46
   |
17 |     let handles_fut = foo_handles_from_locks(locks);
   |                       ---------------------- ^^^^^
   |                       |                      |
   |                       |                      expected `&FooLocks`, found struct `FooLocks`
   |                       |                      help: consider borrowing here: `&locks`
   |                       arguments to this function are incorrect
   |
note: function defined here
  --> tests/server_state/type_names.rs:5:1
   |
5  | #[server_state]
   | -^^^^^^^^^^^^^^
   = note: this error originates in the attribute macro `server_state` (in Nightly builds, run with -Z macro-backtrace for more info)
